File: system-test/sec-tests/java/Sec25Jwt.java
// SEC-25: JWT Security Defects
// Test: TC-CR-031
package com.example;

import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.SignatureAlgorithm;

public class JwtService {
    public String createToken(String username) {
        return Jwts.builder()
            .setSubject(username)
            .signWith(SignatureAlgorithm.NONE, "")  // alg:none — no signature
            .compact();
    }

    public String createTokenWithWeakSecret(String username) {
        return Jwts.builder()
            .setSubject(username)
            .signWith(SignatureAlgorithm.HS256, "secret".getBytes())  // Too weak
            .compact();
    }
}

File: system-test/sec-tests/javascript/sec_25_jwt.js
// SEC-25: JWT Security Defects
// Test: TC-CR-031
const jwt = require('jsonwebtoken');

function createToken(payload) {
  return jwt.sign(payload, '', { algorithm: 'none' });  // alg:none
}

function createWeakToken(payload) {
  return jwt.sign(payload, 'secret', { algorithm: 'HS256' });  // Weak secret
}

function decodeToken(token) {
  return jwt.decode(token, { complete: true });  // No signature verification
}
